mcap futures settlement (#40)
vegaprotocol/research: notes/mcap-futures-markets-settlement.md

# Research note: settling market capitalisation pre-launch token futures

This note is a response to community interest in creating futures markets on the market capitalisation of various tokens prior to token launch.

It describes one way to define when and how to settle such markets given the broad possible number of outcomes and the lack of ahead-of-time certainty as to whether a token will launch and if it does, what the market capitalisation will be at launch.

> The approach is modelled on Credit Default Swaps (CDS) which are derivatives that are valued based on the probability of a company defaulting. Payouts are triggered when a “credit event” occurs, as defined by a set of explicit rules. Analogously, we define the concept of an “airdrop event” that can be similarly used to determine when and at what value to settle points futures.

Reviewer: ```suggestion
The approach is modelled on Credit Default Swaps (CDS) which are derivatives that are valued based on the probability of a company defaulting. Payouts are triggered when a “credit event” occurs, as defined by a set of explicit rules. Analogously, we define the concept of an “token launch” that can be similarly used to determine when and at what value to settle MCAP futures.
```

## Conceptual overview

> Conceptually, we need to define three things to enable fair and orderly settlement of an mcap futures market:

Author: Yea, fixed.

> Conceptually, we need to define three things to enable fair and orderly settlement of an mcap futures market:

Author: ```suggestion
Conceptually, we need to define four things to enable fair and orderly settlement of an mcap futures market:
```

2. The conditions under which the ***token launch*** has been deemed to have occurred, and the market may settle.

3. The ***MCAP Valuation Algorithm*** for calculating the relevant market capitalisation for settlement purposes.

4. The ***futures contract settlemnt price***  is defined as soon as the MCAP valuation algorithm has provided an accepted answer hence the value of 1 futures contract on the vega chain can be obtained.

 

## Detailed definition

### Terms

These terms have specific meanings in the text below

*token*
: Any asset issued by a protocol or company that has utility for the project / company and exists on any public blockchain (a typical example would be an ERC20 on Ethereum). 

*termination date/time*
: The time at which trading on the vega futures market should stop as a result  ***MCAP Valuation Algorithm*** providing an accepted value.

### Variables

The following terms (with suggested values in brackets) used in the definitions refer to quantities or other parameters that affect the specifics of the product. They may be varied to suit each individual use case.

`Base Asset` (USDT)
: The asset in which market capitalisation of the token is considered.

> `Inclusivity Threshold` (40.00%)

Author: Changed.

: The percentage of tokens freely circulating out of a total number of project tokens.

`Expiry Date/Time` (optional)
: *If specified* the market would settle at a price of zero at the expiry date/time if no accepted value has been produced by the ***MCAP Valuation Algorithm*** on or before the specified date and time.

`Liquidity Threshold` (0.05% = 5bp = 0.0005)
: The fraction of total number of tokens that must available to be bought *and* sold in the market (across valid exchanges defined below) within `valid slippage` (%) of the current price per point.

`Valid Slippage` (20.00%)
: The range around the current market price at which available bids are deemed to constitute ‘liquidity’ for the purpose of the ***MCAP Valuation Algorithm***.

> `Minimum Trading Window` (48 hours)

Author: Changed to a week.

> : The minimum amount of cumulative time over which a market meeting the liquidity and slippage thresholds must exist to trigger a succesful ***MCAP Valuation Algorithm*** calculation.

Reviewer 2: ```suggestion
: The minimum amount of cumulative time over which a market meeting the liquidity and slippage thresholds must exist to trigger a successful ***MCAP Valuation Algorithm*** calculation.
```

`Valid Exchanges`
: A list of exchanges on which the listing of the token would count as valid for measurement of price. These may include a list of major centralised exchanges such as Binance, ByBit or Coinbase or significant decentralised spot exchanges on the chains where the token is likely to be launched (Uniswap for Ethereum, Osmosis for Cosmos etc.).

### MCAP Valuation Algorithm

> Once the token has launched and has been trading for `Minimum Trading Window` with sufficient `Liquidity Threshold` the valuation algorithm will calculate the MCAP as the time-weighted average price, over the minimum trading window times the total number of tokens.

Author: Re-wrote.
